src/debug/debug.cc

```diff
@@ -3703,7 +3703,12 @@ static void RecordReplayRegisterScript(Handle<Script> script) {
   // than line zero / column zero, the script must be inlined into another file.
   Script::PositionInfo start_info;
   Script::GetPositionInfo(script, 0, &start_info, Script::WITH_OFFSET);
-  const char* kind = (start_info.line || start_info.column) ? "inlineScript" : "scriptSource";
+
+  // [RUN-2172] Inline scripts sometimes might have line and column but no
+  // URL in case of blink-internal scripts.
+  const char* kind = ((start_info.line || start_info.column) && !url.empty())
```

Comment:
The change seems to contradict the comment (?)

You're saying inline scripts might have "no URL" in some cases, and then in the conditional you are checking `!url.empty()`, which selects "inlineScript" when the URL is non-empty.

Reply:
Good catch. That was badly written. Fixed the comment.